x-pack/test/cases_api_integration/security_and_spaces/tests/common/user_actions/find_user_actions.ts

```diff
@@ -0,0 +1,935 @@
+          expect(createCaseUserAction.type).to.eql('create_case');
+          expect(createCaseUserAction.action).to.eql('create');
+        });
+      });
+
+      describe('rbac', () => {
+        const supertestWithoutAuth = getService('supertestWithoutAuth');
+
+        let secCase: CaseResponse;
+        let obsCase: CaseResponse;
+        let secCaseSpace2: CaseResponse;
+
+        before(async () => {
+          [secCase, obsCase, secCaseSpace2] = await Promise.all([
+            createCase(
+              supertestWithoutAuth,
+              getPostCaseRequest({ owner: 'securitySolutionFixture' }),
+              200,
+              {
+                user: secOnly,
+                space: 'space1',
+              }
+            ),
+            createCase(
+              supertestWithoutAuth,
+              getPostCaseRequest({ owner: 'observabilityFixture' }),
+              200,
+              {
+                user: obsOnly,
+                space: 'space1',
+              }
+            ),
+            createCase(
+              supertestWithoutAuth,
+              getPostCaseRequest({ owner: 'securitySolutionFixture' }),
+              200,
+              {
+                user: superUser,
+                space: 'space2',
+              }
+            ),
+          ]);
+        });
+
+        it('should return with the correct status code when executing with various users', async () => {
+          for (const scenario of [
+            {
+              user: globalRead,
+              id: secCase.id,
+              space: 'space1',
+            },
+            {
+              user: globalRead,
+              id: obsCase.id,
+              space: 'space1',
+            },
+            {
+              user: superUser,
+              id: secCase.id,
+              space: 'space1',
+            },
+            {
+              user: superUser,
+              id: obsCase.id,
+              space: 'space1',
+            },
+            {
+              user: secOnlyRead,
+              id: secCase.id,
+              space: 'space1',
+            },
+            {
+              user: obsOnlyRead,
+              id: obsCase.id,
+              space: 'space1',
+            },
+          ]) {
+            const res = await findCaseUserActions({
+              caseID: scenario.id,
+              supertest: supertestWithoutAuth,
+              options: {
+                sortOrder: 'asc',
+                types: [ActionTypes.create_case],
+              },
+              auth: { user: scenario.user, space: scenario.space },
+            });
+
+            expect(res.userActions.length).to.be(1);
+          }
+        });
+
+        it('should fail to find user actions for a case that the user is not authorized for', async () => {
+          for (const scenario of [
+            {
+              user: secOnlyRead,
+              id: obsCase.id,
+              space: 'space1',
+              expectedCode: 200,
```

Comment:
This is interesting. I am wondering, as this is not a generic find for all user actions but specific for a case if it makes sense to get the case, check if the user is authorized for the case, and throw if not. What do you think? Do we follow this pattern anywhere else?

Reply:
Yeah I could go either way on this. I see what you're saying and I think it'd make sense to throw since a case can only have one type of owner for user actions within the case. The only other example I could find was for finding comments:

https://github.com/elastic/kibana/blob/main/x-pack/test/cases_api_integration/security_and_spaces/tests/common/comments/find_comments.ts#L313

It looks like we have the same behavior as I have written here in that it returns a 200 and finds no results but doesn't throw. Looks like I also wrote those tests though 😆 so maybe they're wrong.

Reply:
Hahaha. Nah, they are as expected. I searched the codebase and we do not use this endpoint, nor have we documented it. I think it makes more sense if we throw if the user does not have access to the case as the route is bound to the case.